[FABG-1008] Permit the msp.SigningIdentity to return the PrivateKey Bytes

pplavetzki opened the pull request:
This PR is to allow the Signing Identity of an ECDSA Key to return the private key portion.  This is to allow a newly enrolled user to create a new X509 Identity.

This is the same workflow that the fabric-sdk-node follows.

Also, this change follows the same logic of the AES Key, using an exportable flag to verify if the Bytes() method should be supported or not.

This is in response to the jira issue I created: [FABG-1008](https://jira.hyperledger.org/browse/FABG-1008)

bstasyszyn committed:
[FABG-994] Implement lifecycle query approved chaincode (#113)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-995] Implement lifecycle check commit readiness (#114)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-996] Lifecycle commit chaincode (#115)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-997] Lifecycle query committed chaincode (#116)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

troyronda committed:
[FABG-1005] chaincoded should use distinct separator (#125)

Chaincoded removes the last segment of the chaincode ID when detecting the binary name.
The last segment separator is changed to the more distinct _fabtest_.

Signed-off-by: Troy Ronda <troy@troyronda.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-1006] Fix for query committed "responses do not match" (#129)

Compare the protos in the responses instead of the bytes.

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

huangjiehua committed:
[FABG 998]Integration tests (#136)

* Integration tests include: package cc, install cc, get installed cc package, query installed cc, approve cc, query approve cc, check commit readiness, commit cc, query committed cc

Signed-off-by: Jiehua Huang <huangjiehua@aisino.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

huangjiehua committed:
fix lint-submodules errors (#137)

* fix lint-submodules errors

Signed-off-by: Jiehua Huang <huangjiehua@aisino.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

troyronda committed:
[FABG-1007] Release v1.0.0-beta3 (#140)

Signed-off-by: Troy Ronda <troy@troyronda.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
added an exportable field to the struct to allow Bytes method

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
added an exportable field to the struct to allow Bytes method

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
updated to respect the exportable flag

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
updated to respect the exportable flag

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
allowing the exporting of the private key

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-1009] Wait for responses from other Discovery targets (#141)

This commit fixes a problem where the iFabric selection service returns immediately after receiving a transient error from a target peer, even though another peer may have replied successfully.

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-1008] This is to is to allow the Signing Identity of an ECDSA Key to return the private key portion.
This is to allow a newly enrolled user to create a new X509 Identity.

[FABG-997] Lifecycle query committed chaincode (#116)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG-1005] chaincoded should use distinct separator (#125)

Chaincoded removes the last segment of the chaincode ID when detecting the binary name.
The last segment separator is changed to the more distinct _fabtest_.

Signed-off-by: Troy Ronda <troy@troyronda.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG-1006] Fix for query committed "responses do not match" (#129)

Compare the protos in the responses instead of the bytes.

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG 998]Integration tests (#136)

* Integration tests include: package cc, install cc, get installed cc package, query installed cc, approve cc, query approve cc, check commit readiness, commit cc, query committed cc

Signed-off-by: Jiehua Huang <huangjiehua@aisino.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

fix lint-submodules errors (#137)

* fix lint-submodules errors

Signed-off-by: Jiehua Huang <huangjiehua@aisino.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG-1007] Release v1.0.0-beta3 (#140)

Signed-off-by: Troy Ronda <troy@troyronda.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

added an exportable field to the struct to allow Bytes method

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

updated to respect the exportable flag

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

allowing the exporting of the private key

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG-1009] Wait for responses from other Discovery targets (#141)

This commit fixes a problem where the iFabric selection service returns immediately after receiving a transient error from a target peer, even though another peer may have replied successfully.

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

[FABG-1010] Added ConnectionFailed to ResMgmtDefaultRetryableCodes (#142)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

returning an error if the Marshal process fails

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

added an exportable field to the struct to allow Bytes method

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

updated to respect the exportable flag

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

bstasyszyn committed:
[FABG-1010] Added ConnectionFailed to ResMgmtDefaultRetryableCodes (#142)

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>
Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
returning an error if the Marshal process fails

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
Merge branch 'master' into patch/ecdsakey-bytes

troyronda commented:
@pplavetzki Thanks for your contribution. Would it be possible to squash your commits and include [FABG-1008] in your commit description.

bstasyszyn committed:
[FABG-1012] Retry on private data dissemination error (#144)

The channel client now retries on error "failed to distribute private collection" since this is most likely a transient error.

Signed-off-by: Bob Stasyszyn <Bob.Stasyszyn@securekey.com>

pplavetzki committed:
[FABG-1008] Allow the Signing Identity of an ECDSA Key to return the private key portion. This is to allow a newly enrolled user to create a new X509 Identity.

pplavetzki committed:
Merge branch 'master' into patch/ecdsakey-bytes

pplavetzki committed:
fixed an merge error when rebasing

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>

pplavetzki committed:
Merge branch 'patch/ecdsakey-bytes' of https://github.com/pplavetzki/fabric-sdk-go into patch/ecdsakey-bytes

pplavetzki committed:
Merge branch 'patch/ecdsakey-bytes' of https://github.com/pplavetzki/fabric-sdk-go into patch/ecdsakey-bytes

pplavetzki commented:
@troyronda I tried rebasing to squash, but I think I made a royal mess of this PR.  Any guidance on how to properly do this.  Many thanks, cheers!

alikic commented:
It would be good to add some test cases for this new functionality. As for fixing the PR, as the last resort, you can always drop this PR and create a clean one.

pplavetzki committed:
added a unit test to verify PrivateKey.Bytes() functionality

Signed-off-by: Paul Plavetzki <paul@pareidolia.io>
